clarity_ext/repository/step_repository.py

```diff
@@ -44,16 +45,10 @@ def all_artifacts(self):
         instead.
         """
         input_output_maps = self.session.current_step.api_resource.input_output_maps
-        artifact_keys = set()
-        for input, output in input_output_maps:
-            artifact_keys.add(input["uri"])
-            artifact_keys.add(output["uri"])
-
-        artifacts = self.session.api.get_batch(artifact_keys)
-        artifacts_by_uri = {artifact.uri: artifact for artifact in artifacts}
-        for input, output in input_output_maps:
-            input['uri'] = artifacts_by_uri[input['uri'].uri]
-            output['uri'] = artifacts_by_uri[output['uri'].uri]
+        wrappable_pairs = WrappablePairs(self.session, input_output_maps, WrappablePairs.MODE_STATELESS)
+        # We have to defer raising error until all artifacts are fetched.
+        # Step log is in the artifacts. The errors are to be written into the step log...
+        self.xml_discordance_errors = wrappable_pairs.validate()
```

Comment:
Related to: https://github.com/Molmed/clarity-ext/pull/209/files#r608413175

I would prefer if we would simply throw an error here if the validation doesn't work. E.g. a `AssertionError`

Reply:
Maybe you don't do that because of that thing with `Step log` not being loaded yet? Correct me if I'm wrong :)

If that's the case: I don't think we need to worry about that because this error is not intended to be read by the end user but rather by us. We absolutely don't expect this to happen so in my mind it's enough that it just ends up in the system log. The step log was originally intended just for errors that are interesting to the user.

Reply:
Sounds good!